### PR TITLE
Separate cf cli environment #89

### PR DESCRIPTION
**Pull request by atwoosnam**
Add "CF_HOME" env var to all CF CLI commands; create a new `config.json` file at whatever directory is specified by the VSIX package.

### CODE REVIEW

**Comment by atwoosnam** [comment]
See [comment in issue #89](https://github.com/vmware-tanzu/tanzu-toolkit-for-visual-studio/issues/89#issuecomment-870064965) for demo vid

File: src/Services/src/CmdProcess/ICmdProcessService.cs [hunk 1]
```diff
@@ -1,9 +1,10 @@
-﻿using static Tanzu.Toolkit.Services.OutputHandler.OutputHandler;
+﻿using System.Collections.Generic;
+using static Tanzu.Toolkit.Services.OutputHandler.OutputHandler;
 
 namespace Tanzu.Toolkit.Services.CmdProcess
 {
     public interface ICmdProcessService
     {
-        CmdResult RunCommand(string executableFilePath, string arguments, string workingDir, StdOutDelegate stdOutDelegate, StdErrDelegate stdErrDelegate);
+        CmdResult RunCommand(string executableFilePath, string arguments, string workingDir, StdOutDelegate stdOutDelegate, StdErrDelegate stdErrDelegate, Dictionary<string, string> envVars = null);
```
Comment (ccheetham) [inline src/Services/src/CmdProcess/ICmdProcessService.cs:8]: I think the order of params might be better as:
* exe
* args
* workdir
* env
* stdout
* stderr

The reason being that the "input" vars would be the leading params and the "output" params would be the trailing params.

File: src/Services/src/CmdProcess/CmdProcessService.cs [hunk 2]
```diff
@@ -25,6 +26,15 @@ public CmdResult RunCommand(string executableFilePath, string arguments, string
                 process.StartInfo.WorkingDirectory = workingDir;
             }
 
+            // Set environment variables
+            if (envVars != null)
+            {
+                foreach (string key in envVars.Keys)
```
Comment (ccheetham) [inline src/Services/src/CmdProcess/CmdProcessService.cs:32]: A common idiom for iterating dictionary values is using a tuple ...
```
foreach (var envVar in envVars)
{
    process.StartInfo.EnvironmentVariables[envVar.Key] = envVars[envVar.Value];
```

File: src/Services/src/CfCli/CfCliService.cs [hunk 3]
```diff
@@ -563,8 +567,12 @@ public DetailedResult ExecuteCfCliCommand(string arguments, string workingDir =
                 return new DetailedResult(false, _cfExePathErrorMsg);
             }
 
+            var envVars = new Dictionary<string, string> {
```
Comment (ccheetham) [inline src/Services/src/CfCli/CfCliService.cs:570]: should `{` be on the next line?

**Review by ccheetham — CHANGES_REQUESTED** [review]

File: src/Services/src/CmdProcess/ICmdProcessService.cs (hunk 1 above)
Comment (atwoosnam) [reply src/Services/src/CmdProcess/ICmdProcessService.cs:8]: Fair point - happy to change the order but generally I like to have optional params come after required ones...
How's this? (made the delegates optional as well so they can come last)
`CmdResult RunCommand(string executableFilePath, string arguments, string workingDir, Dictionary<string, string> envVars = null, StdOutDelegate stdOutDelegate = null, StdErrDelegate stdErrDelegate = null);
`

**Review by ccheetham — APPROVED** [review]
lgtm

**Pull request merged**